### PR TITLE
fix(upload): fixed json bug on resume upload

### PR DESCRIPTION
**Pull request by cnlangzi**
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master BEFORE merging this PR

Associated PRs (Link as appropriate):
- zbox: https://github.com/0chain/zboxcli/pull/164
- blobber: https://github.com/0chain/blobber/pull/533

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/resume_bug_on_json  |
| zboxcli  | fix/resume_bug_on_json  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: core/util/compact_merkle_tree_test.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ func TestCompactMerkleTreeWithEvenLeaves(t *testing.T) {
 					    [2+3]─────┘               │
 					  3──┘                        │
 				                    [[[0+1]+[2+3]]+[[4+5]+[4+5]]]
-				  	  4──┐                        │
+				  	4──┐                        │
```
Comment (princeparmar) [inline core/util/compact_merkle_tree_test.go:21]: I think some typing mistake

File: zboxcore/sdk/chunked_upload_model.go [hunk 2]
```diff
@@ -113,6 +116,30 @@ type UploadBlobberStatus struct {
...
+func (s *UploadBlobberStatus) UnmarshalJSON(b []byte) error {
+	//fixed Hasher doesn't work in UnmarshalJSON
+	status := &status{}
+
+	if err := json.Unmarshal(b, status); err != nil {
+		return err
+	}
+
+	status.Hasher.File = sha256.New()
+	status.Hasher.Content.Hash = func(left, right string) string {
```
Comment (princeparmar) [inline zboxcore/sdk/chunked_upload_model.go]: accessing Content.Hash without checking content is nil or not may create nil pointer err.

File: zboxcore/sdk/chunked_upload_model.go [hunk 3]
```diff
@@ -113,6 +116,30 @@ type UploadBlobberStatus struct {
...
+		return encryption.Hash(left + right)
+	}
+
+	s.Hasher = &status.Hasher
```
Comment (princeparmar) [inline zboxcore/sdk/chunked_upload_model.go:143]: can s also be nil as it is pointer?

**Review by princeparmar — COMMENTED** [review]
I am not much familiar with this logic still i have checked the code and seem good. Some minor changes i have added.

File: core/util/compact_merkle_tree_test.go (hunk 1 above)
Comment (cnlangzi) [reply core/util/compact_merkle_tree_test.go:21]: do you mean what is the mistake?

Comment (cnlangzi) [reply core/util/compact_merkle_tree_test.go:21]: see detail on https://github.com/0chain/blobber/wiki/Protocols#what-is-compactmerkletree

Comment (princeparmar) [reply core/util/compact_merkle_tree_test.go:21]: one space removed because of which it is not proper tree. this change was not required i think.

**Review by princeparmar — APPROVED** [review]

**Pull request merged**